client/components/common/UserCustomSets.tsx

```diff
@@ -0,0 +1,466 @@
+    <Select<string>
+      getPopupContainer={(node: HTMLElement) => {
+        if (node.parentElement) {
+          return node.parentElement;
+        }
+        return document && document.body;
+      }}
+      css={[
+        { ...inputFontSize },
+        { marginTop: 20, [mq[1]]: { marginTop: 0, flex: 1 } },
+      ]}
+      showSearch
+      filterOption={(input, option) => {
+        return (option?.children[1] as string)
+          .toLocaleUpperCase()
+          .includes(input.toLocaleUpperCase());
+      }}
+      value={dofusClassId}
+      onChange={(value: string) => {
+        setDofusClassId(value);
+      }}
+      placeholder={t('SELECT_CLASS')}
+      allowClear
+    >
+      {[...classesData.classes]
+        .sort(({ name: n1 }, { name: n2 }) => n1.localeCompare(n2))
+        .map((dofusClass) => (
+          <Select.Option key={dofusClass.id} value={dofusClass.id}>
+            <img
+              src={dofusClass.faceImageUrl}
+              alt={dofusClass.name}
+              css={{ width: 20, marginRight: 8 }}
+            />
+            {dofusClass.name}
+          </Select.Option>
+        ))}
+    </Select>
+  );
+
+  return (
+    <div
+      css={{
+        display: 'flex',
+        flexDirection: 'column',
+        maxWidth: 1036,
+        width: '100%',
+        margin: '0 auto',
+        [mq[1]]: { flexDirection: 'row' },
+      }}
+    >
+      <div
+        css={{
+          display: 'flex',
+          flexDirection: 'row',
+          margin: '0px 20px 20px 0px',
+          gridArea: '1 / 3 / 1 /1',
+          borderRadius: 6,
+          [mq[1]]: {
+            flexDirection: 'column',
+            margin: '10px 0px 0px 0px',
+            padding: '0px 20px 20px 20px',
+            width: '25%',
+          },
+        }}
+      >
+        <img
+          src="https://media.discordapp.net/attachments/645410912605437972/924539234709282866/Layer_9.png"
+          alt="Avatar"
+          css={{
+            maxWidth: 120,
+            alignItems: 'center',
+            borderRadius: 6,
+            border: '4px solid black',
+            outline: '1px solid #434343',
+            [mq[1]]: {
+              maxWidth: 240,
+              alignItems: 'flex-start',
+            },
+          }}
+        />
+        <div
+          css={{
+            display: 'flex',
+            flexDirection: 'column',
+            justifyContent: 'center',
+            marginLeft: 20,
+            [mq[1]]: {
+              marginLeft: 0,
+            },
+          }}
+        >
+          <h1 css={{ margin: '10px 0px 0px 0px', fontSize: 28 }}>{`${capitalize(
+            username as string,
+          )}`}</h1>
+          <span css={{ color: '#8b949e' }}>Member since 05/01/2021</span>
+        </div>
+      </div>
+      <Tabs
+        defaultActiveKey="1"
+        onChange={() => {}}
+        css={{ gridArea: '4/1/4/3', width: '100%', padding: 10 }}
+      >
+        <TabPane tab="Builds" key="1">
+          <InfiniteScroll
```

Comment:
Seems like a very large chunk of this page is the same as the MyBuilds component. Can we share more code between them? Maybe have a <Builds /> component that takes in the query result and an onLoadMore as a prop?